XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -400,14 +399,46 @@ private void UpdateDetails()
 
         }
 
-        private static string RecommendedPatchesForHost(Host host)
+        private static string RequiredUpdatesForHost(Host host)
         {
             var result = new List<string>();
-            var recommendedPatches = Updates.RecommendedPatchesForHost(host);
+            var requiredUpdates = Updates.RecommendedPatchesForHost(host);
 
-            foreach (var patch in recommendedPatches)
+            if (requiredUpdates == null)
+            {
+                // versions with no minimum patches
+                var updatesList = new List<string>();
+                var alerts = new List<Alert>(Updates.UpdateAlerts);
+                if (alerts.Count == 0)
+                    return String.Empty;
+
+                foreach (Alert alert in alerts)
+                {
+                    if (!(alert is XenServerPatchAlert))
+                        continue;
+                    if (alert.AppliesTo.Contains(host.Name))
+                        updatesList.Add(alert.Name);
+                }
+
+                updatesList.Sort(StringUtility.NaturalCompare);
+                return string.Join(", ", updatesList.ToArray());
+            }
+

```

Comment:
It is good as is, but for better maintainability I'd suggest you put the rest in an else branch, so it will make sure that the return in line 424 can't just be removed and also this would make the code a lot more readable.
Also, the 'result' List is only needed here, let's move it here.

```
if ()
{
unchanged
return unchanged
}
else //new
{
var result....
everything else
return
}
```